lib/runner.js

```diff
@@ -418,17 +421,40 @@ Runner.prototype.hook = function(name, fn) {
         }
       } else if (err) {
         self.failHook(hook, err);
-        // stop executing hooks, notify callee of hook err
-        return fn(err);
+
+        if (name === HOOK_TYPE_BEFORE_ALL) {
+          suite.tests.forEach(function(test) {
```

Comment:
would prefer a method on `Suite.prototype` which marked all of its tests (and child suites) as `skipped`, rather than digging into the properties on a `Suite` instance